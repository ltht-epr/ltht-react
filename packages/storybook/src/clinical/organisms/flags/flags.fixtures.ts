import { Flag, FlagStatusCode, PartialDateTimeKindCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: false,
  requestedWhen: '',
}

const redactedMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: true,
  requestedWhen: '',
}

const FlagOne: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe5|personalertandhazard',
  isEnteredInError: false,
  metadata: redactedMetadata,
  author: {
    display: 'Smith, John (Dr)',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'Lives in Known High Risk Area', code: '109006' }],
  },
  category: {
    coding: [
      { display: 'Location', code: '66706' },
      { display: 'Demographics', code: '66707' },
    ],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Year,
    },
  },
  status: FlagStatusCode.EnteredInError,
  text: {
    div: 'Lorem ipsum dolor sit amet',
    text:
      'Lorem ipsum dolor sit amet, falli nulla oportere vel eu. Ea meis moderatius scripserit his, ei decore ceteros has. Quo at apeirian facilisi consectetuer. At putant animal persius his, inani maiestatis moderatius vis no',
  },
}

const FlagTwo: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard',
  isEnteredInError: false,
  metadata: mockMetadata,
  author: {
    display: 'John Smith',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'High Risk of Violence', code: '56706' }],
  },
  category: {
    coding: [{ display: 'Violence', code: '66706' }],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },

  status: FlagStatusCode.Inactive,
}

const flags: Flag[] = [FlagOne, FlagTwo]

export default flags
