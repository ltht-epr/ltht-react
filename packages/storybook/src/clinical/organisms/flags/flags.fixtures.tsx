import { ButtonProps } from '@ltht-react/button'
import Icon from '@ltht-react/icon'
import { Flag, FlagStatusCode, PartialDateTimeKindCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: false,
  requestedWhen: '',
}

const redactedMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: true,
  requestedWhen: '',
}

const FlagOne: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe5|personalertandhazard',
  isEnteredInError: false,
  metadata: mockMetadata,
  author: {
    display: 'Smith, John (Dr)',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'DNACPR', code: '109007' }],
  },
  category: {
    coding: [
      { display: 'Location', code: '66706' },
      { display: 'Demographics', code: '66707' },
    ],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Year,
    },
  },
  status: FlagStatusCode.Active,
  text: {
    div: 'Lorem ipsum dolor sit amet',
    text: 'A ReSPECT plan for emergency care, including a DO NOT ATTEMPT CARDIOPULMONARY RESUSCITATION decision, is in place. Please review the ReSPECT form and CPR decision, check for changes and, if valid and applicable, take this DNACPR status into account.',
  },
  extension: [
    { url: 'https://leedsth.nhs.uk/alert/priority', valueString: 'High' },
    { url: 'https://leedsth.nhs.uk/alert/advice', valueString: 'Some advice' },
  ],
}

const FlagTwo: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard',
  isEnteredInError: false,
  metadata: mockMetadata,
  author: {
    display: 'John Smith',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'Oxygen Alert', code: 'BLEOMYCINM' }],
  },
  category: {
    coding: [{ display: 'Violence', code: '66706' }],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: FlagStatusCode.Active,
  text: {
    div: 'Lorem ipsum dolor sit amet',
    text: 'It has been confirmed that Bleomycin has previously been administered to this patient. They may experience severe lung toxicity or death if high concentration oxygen is administered. Commence oxygen therapy only if SaO2 <85%. SaO2 should be maintained no higher than 85-88% whether on invasive or non invasive ventilation. Refer to LTHT Bleomycin pulmonary toxicity guidelines on LHP for further information/contact the Oncology enquiries nurse bleepholder via LTHT switchboard on 0113 2433144.',
  },
  extension: [{ url: 'https://leedsth.nhs.uk/alert/priority', valueString: 'High' }],
}

const FlagThree: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard',
  isEnteredInError: false,
  metadata: mockMetadata,
  author: {
    display: 'John Smith',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'Adrenal Insufficiency', code: 'STEROID' }],
  },
  category: {
    coding: [{ display: 'Violence', code: '66706' }],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: FlagStatusCode.Active,
  text: {
    div: 'Lorem ipsum dolor sit amet',
    text: 'Requirement for Urgent Glucocorticoids',
  },
  extension: [
    { url: 'https://leedsth.nhs.uk/alert/priority', valueString: 'High' },
    {
      url: 'https://leedsth.nhs.uk/alert/advice',
      valueString:
        'This patient has adrenal insufficiency and is steroid dependent. Any serious injury or illness may precipitate an adrenal crisis. In the event of a serious illness/accident they require the following age-related doses of Hydrocortisone immediately by IM or IV injection: Age <1 year: Hydrocortisone 25 mg Age 1-5 years: Hydrocortisone 50 mg Age > 6 years: Hydrocortisone 100 mg The usual oral steroids should be doubled or tripled. Please seek advice from the on-call endocrinologist',
    },
    {
      url: 'https://leedsth.nhs.uk/alert/linkeddocumentsource',
      valueString: 'https://www.edge.nhs.uk/#/project/100005/site/198655/details',
    },
  ],
}

const FlagFour: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard',
  isEnteredInError: false,
  metadata: mockMetadata,
  author: {
    display: 'John Smith',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'Life Threatening Anaphylaxis', code: 'ANAPHYLAXI' }],
  },
  category: {
    coding: [{ display: 'Violence', code: '66706' }],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: FlagStatusCode.Active,
  text: {
    div: 'Lorem ipsum dolor sit amet',
    text: 'This patient has a history of life threatening anaphylaxis.',
  },
  extension: [{ url: 'https://leedsth.nhs.uk/alert/priority', valueString: 'High' }],
}

const FlagFive: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard',
  isEnteredInError: false,
  metadata: mockMetadata,
  author: {
    display: 'John Smith',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'VIOLENT PATIENT', code: 'VIOLENT' }],
  },
  category: {
    coding: [{ display: 'Violence', code: '66706' }],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: FlagStatusCode.Active,
  text: {
    div: 'Lorem ipsum dolor sit amet',
    text: 'This patient has a HISTORY OF VIOLENCE TOWARDS STAFF / OTHERS ',
  },
  extension: [{ url: 'https://leedsth.nhs.uk/alert/priority', valueString: 'High' }],
}

const FlagSix: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe5|personalertandhazard',
  isEnteredInError: false,
  metadata: mockMetadata,
  author: {
    display: 'Smith, John (Dr)',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'ReSPECT', code: 'ReSPECT' }],
  },
  category: {
    coding: [
      { display: 'Location', code: '66706' },
      { display: 'Demographics', code: '66707' },
    ],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Year,
    },
  },
  status: FlagStatusCode.Inactive,
  text: {
    div: 'Lorem ipsum dolor sit amet',
    text: 'A ReSPECT plan for care in emergency situations has been completed. Please read and consider these recommendations in event of clinical deterioration.',
  },
  extension: [{ url: 'https://leedsth.nhs.uk/alert/priority', valueString: 'Clinical' }],
}

const FlagSeven: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard',
  isEnteredInError: false,
  metadata: redactedMetadata,
  author: {
    display: 'John Smith',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'Infection', code: 'MRSA' }],
  },
  category: {
    coding: [{ display: 'Violence', code: '66706' }],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: FlagStatusCode.Inactive,
  text: {
    div: 'Infection Prevention & Control Hazard, please use Apron & Gloves',
    text: 'Staphylococcus aureus - MRSA',
  },
}

const FlagEight: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard',
  isEnteredInError: false,
  metadata: mockMetadata,
  author: {
    display: 'John Smith',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'Infection', code: 'VRE' }],
  },
  category: {
    coding: [{ display: 'Violence', code: '66706' }],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: FlagStatusCode.Inactive,
  text: {
    div: 'Infection Prevention & Control Hazard, please use Apron & Gloves',
    text: 'Vancomycin-resistant Enterococcus sp',
  },
}

const FlagNine: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard',
  isEnteredInError: false,
  metadata: mockMetadata,
  author: {
    display: 'John Smith',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'Active Clinical Trial', code: 'TRIALALERT' }],
  },
  category: {
    coding: [{ display: 'Violence', code: '66706' }],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: FlagStatusCode.Inactive,
  text: {
    div: 'he patient is on an active trial and if the patient is acutely admitted then the Principal Investigator and Trial Nurse need to be informed within 24 hours of the admission.',
    text: 'SOMAscan Diabetes',
  },
}

const FlagTen: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe5|personalertandhazard',
  isEnteredInError: false,
  metadata: mockMetadata,
  author: {
    display: 'Smith, John (Dr)',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'Infection', code: 'KPNE' }],
  },
  category: {
    coding: [
      { display: 'Location', code: '66706' },
      { display: 'Demographics', code: '66707' },
    ],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Year,
    },
  },
  status: FlagStatusCode.Active,
  text: {
    div: 'Infection Prevention & Control Hazard, please use Apron & Gloves',
    text: 'Klebsiella pneumoniae',
  },
}

const FlagEleven: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe5|personalertandhazard',
  isEnteredInError: false,
  metadata: mockMetadata,
  author: {
    display: 'Smith, John (Dr)',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'Data Quality', code: 'DQMERGEDUP' }],
  },
  category: {
    coding: [
      { display: 'Location', code: '66706' },
      { display: 'Demographics', code: '66707' },
    ],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Year,
    },
  },
  status: FlagStatusCode.EnteredInError,
  text: {
    div: 'There is another record in the system which is probably for this patient. Please raise this issue to the Informatics Service Desk and supply the patient details.',
    text: 'Possible Duplicate Patient Exists',
  },
}

const flags: Flag[] = [
  FlagOne,
  FlagThree,
  FlagTwo,
  FlagFour,
  FlagFive,
  FlagSix,
  FlagSeven,
  FlagEight,
  FlagNine,
  FlagTen,
  FlagEleven,
]

const controls: ButtonProps[] = [
  {
    icon: <Icon type="paper-clip" size="medium" />,
    value: 'Control 1',
    title: 'Description of the button action can go here!!!',
    styling: {
      buttonStyle: 'workflow',
      padding: '20px',
    },
  },
  {
    icon: <Icon type="link" size="medium" />,
    value: 'Control 2',
    styling: {
      buttonStyle: 'standard',
      padding: '2px 10px',
    },
  },
  {
    icon: <Icon type="trash" size="medium" />,
    value: 'Control 3',
    styling: {
      buttonStyle: 'danger',
      padding: '2px 10px',
    },
  },
  {
    icon: <Icon type="star" size="medium" />,
    value: 'Control 4',
    styling: {
      buttonStyle: 'primary',
      padding: '2px 10px',
    },
  },
  {
    icon: <Icon type="check" size="medium" />,
    iconColour: 'black',
    value: 'Control 5',
    styling: {
      buttonStyle: 'clear',
      padding: '2px 10px',
    },
  },
  {
    value: 'Click Me!',
    styling: {
      buttonStyle: 'standard',
      padding: '2px 10px',
    },
    onClick: (e) => {
      e.stopPropagation()
      // eslint-disable-next-line no-alert
      alert('clicked me!')
    },
  },
]

export { flags, controls }
