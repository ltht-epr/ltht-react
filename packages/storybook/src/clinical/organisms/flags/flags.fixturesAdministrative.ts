import { Flag, FlagStatusCode, PartialDateTimeKindCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: false,
  requestedWhen: '',
}

// const redactedMetadata: Metadata = {
//   dataSources: [
//     {
//       display: 'Mock',
//     },
//   ],
//   isRedacted: true,
//   requestedWhen: '',
// }

const FlagOne: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe5|personalertandhazard',
  isEnteredInError: false,
  metadata: mockMetadata,
  author: {
    display: 'Smith, John (Dr)',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    typeName: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'Data Quality', code: 'DQMERGEDUP' }],
  },
  category: {
    coding: [
      { display: 'Location', code: '66706' },
      { display: 'Demographics', code: '66707' },
    ],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Year,
    },
  },
  status: FlagStatusCode.EnteredInError,
  text: {
    div:
      'There is another record in the system which is probably for this patient. Please raise this issue to the Informatics Service Desk and supply the patient details.',
    text: 'Possible Duplicate Patient Exists',
  },
}

const flagsAdministrative: Flag[] = [FlagOne]

export default flagsAdministrative
