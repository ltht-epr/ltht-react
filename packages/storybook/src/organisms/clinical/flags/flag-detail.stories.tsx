import React from 'react'
import { storiesOf } from '@storybook/react'
import JSXAddon from 'storybook-addon-jsx'

import FlagDetail from '@ltht-react/flag-detail'
import readme from '@ltht-react/flag-detail/README.md'
import { Card, CardHeader, CardBody } from '@ltht-react/card'

import flags from './flag.fixtures'

const stories = storiesOf('Organisms - Clinical|Flag', module) as any

stories.addWithJSX = JSXAddon.addWithJSX

stories.addParameters({
  readme: {
    sidebar: readme,
  },
})

stories.addWithJSX('Detail', () => (
  <Card noData={!flags}>
    <CardHeader position="center">
      <h3>Flag</h3>
    </CardHeader>
    <CardBody>
      <FlagDetail flag={flags[0]} />
    </CardBody>
  </Card>
))
