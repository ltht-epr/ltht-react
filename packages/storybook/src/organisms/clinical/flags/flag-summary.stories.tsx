import React from 'react'
import { storiesOf } from '@storybook/react'
import JSXAddon from 'storybook-addon-jsx'

import FlagSummary from '@ltht-react/flag-summary'
import readme from '@ltht-react/flag-summary/README.md'
import Card from '@ltht-react/card'
import flags from './flag.fixtures'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const stories = storiesOf('Organisms - Clinical|Flag', module) as any

stories.addWithJSX = JSXAddon.addWithJSX

stories.addParameters({
  readme: {
    sidebar: readme,
  },
})

stories.addWithJSX('Summary', () => (
  <Card>
    <Card.Header>
      <h3>Flags</h3>
    </Card.Header>
    <Card.List>
      {flags.map(flag => (
        <Card.ListItem key={flag.id}>
          <FlagSummary flag={flag} />
        </Card.ListItem>
      ))}
    </Card.List>
  </Card>
))
