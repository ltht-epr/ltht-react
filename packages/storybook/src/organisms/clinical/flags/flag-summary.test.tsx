import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import FlagSummary from '@ltht-react/flag-summary'
import flags from './flag.fixtures'

const Sut: React.FC = () => {
  return (
    <>
      {flags.map(flag => (
        <div key={flag.id}>
          <FlagSummary flag={flag} />
        </div>
      ))}
    </>
  )
}

describe('FlagSummary', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Sut />, div)
  })
  it('matches snapshot', () => {
    expect(mount(<Sut />)).toMatchSnapshot('wrapper mount')
  })
})
