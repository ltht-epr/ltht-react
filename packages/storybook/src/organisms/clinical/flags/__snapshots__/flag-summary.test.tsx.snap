// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlagSummary matches snapshot: wrapper mount 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: left;
}

.emotion-0 {
  color: #768692;
}

.emotion-3 {
  text-align: right;
}

.emotion-2 {
  color: #231f20;
}

.emotion-7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-5 {
  color: #231f20;
  text-align: left;
}

.emotion-6 {
  color: #768692;
  text-align: left;
  font-size: smaller;
  padding-top: 0.25rem;
}

.emotion-10 {
  text-align: right;
}

.emotion-9 {
  color: #768692;
  text-align: right;
  font-size: smaller;
  padding-top: 0.25rem;
}

<Sut>
  <div
    key="580ca927-34e0-e911-a2c7-005056926fe5|personalertandhazard"
  >
    <FlagSummary
      flag={
        Object {
          "author": Object {
            "display": "Smith, John (Dr)",
            "identifier": undefined,
            "reference": "https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser",
            "typeName": "Practitioner",
          },
          "category": Object {
            "coding": Array [
              Object {
                "code": "66706",
                "display": "Location",
              },
              Object {
                "code": "66707",
                "display": "Demographics",
              },
            ],
          },
          "code": Object {
            "coding": Array [
              Object {
                "code": "109006",
                "display": "Lives in Known High Risk Area",
              },
            ],
          },
          "id": "580ca927-34e0-e911-a2c7-005056926fe5|personalertandhazard",
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": true,
            "requestedWhen": "",
          },
          "period": Object {
            "start": Object {
              "kind": "YEAR",
              "value": "2013-01-01T00:00:00+00:00",
            },
          },
          "status": "ENTERED_IN_ERROR",
          "text": Object {
            "div": "Lorem ipsum dolor sit amet",
            "text": "Lorem ipsum dolor sit amet, falli nulla oportere vel eu. Ea meis moderatius scripserit his, ei decore ceteros has. Quo at apeirian facilisi consectetuer. At putant animal persius his, inani maiestatis moderatius vis no",
          },
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-4"
        >
          <FlagRedacted
            flag={
              Object {
                "author": Object {
                  "display": "Smith, John (Dr)",
                  "identifier": undefined,
                  "reference": "https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser",
                  "typeName": "Practitioner",
                },
                "category": Object {
                  "coding": Array [
                    Object {
                      "code": "66706",
                      "display": "Location",
                    },
                    Object {
                      "code": "66707",
                      "display": "Demographics",
                    },
                  ],
                },
                "code": Object {
                  "coding": Array [
                    Object {
                      "code": "109006",
                      "display": "Lives in Known High Risk Area",
                    },
                  ],
                },
                "id": "580ca927-34e0-e911-a2c7-005056926fe5|personalertandhazard",
                "metadata": Object {
                  "dataSources": Array [
                    Object {
                      "display": "Mock",
                    },
                  ],
                  "isRedacted": true,
                  "requestedWhen": "",
                },
                "period": Object {
                  "start": Object {
                    "kind": "YEAR",
                    "value": "2013-01-01T00:00:00+00:00",
                  },
                },
                "status": "ENTERED_IN_ERROR",
                "text": Object {
                  "div": "Lorem ipsum dolor sit amet",
                  "text": "Lorem ipsum dolor sit amet, falli nulla oportere vel eu. Ea meis moderatius scripserit his, ei decore ceteros has. Quo at apeirian facilisi consectetuer. At putant animal persius his, inani maiestatis moderatius vis no",
                },
              }
            }
          >
            <div
              className="emotion-1"
            >
              <RedactedDescription>
                <div
                  className="emotion-0"
                >
                  Insufficient privileges
                </div>
              </RedactedDescription>
            </div>
            <div
              className="emotion-3"
            >
              <PeriodSummary
                period={
                  Object {
                    "start": Object {
                      "kind": "YEAR",
                      "value": "2013-01-01T00:00:00+00:00",
                    },
                  }
                }
              >
                <div
                  className="emotion-2"
                >
                  2013 to Present
                </div>
              </PeriodSummary>
            </div>
          </FlagRedacted>
        </div>
      </Styled(div)>
    </FlagSummary>
  </div>
  <div
    key="580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard"
  >
    <FlagSummary
      flag={
        Object {
          "author": Object {
            "display": "John Smith",
            "identifier": undefined,
            "reference": "https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser",
            "typeName": "Practitioner",
          },
          "category": Object {
            "coding": Array [
              Object {
                "code": "66706",
                "display": "Violence",
              },
            ],
          },
          "code": Object {
            "coding": Array [
              Object {
                "code": "56706",
                "display": "High Risk of Violence",
              },
            ],
          },
          "id": "580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard",
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": false,
            "requestedWhen": "",
          },
          "period": Object {
            "end": Object {
              "kind": "DATE",
              "value": "2014-02-02T00:00:00+00:00",
            },
            "start": Object {
              "kind": "DATE",
              "value": "2013-01-01T00:00:00+00:00",
            },
          },
          "status": "INACTIVE",
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-4"
        >
          <Styled(div)>
            <div
              className="emotion-7"
            >
              <FlagTitle
                flag={
                  Object {
                    "author": Object {
                      "display": "John Smith",
                      "identifier": undefined,
                      "reference": "https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser",
                      "typeName": "Practitioner",
                    },
                    "category": Object {
                      "coding": Array [
                        Object {
                          "code": "66706",
                          "display": "Violence",
                        },
                      ],
                    },
                    "code": Object {
                      "coding": Array [
                        Object {
                          "code": "56706",
                          "display": "High Risk of Violence",
                        },
                      ],
                    },
                    "id": "580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "end": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                      "start": Object {
                        "kind": "DATE",
                        "value": "2013-01-01T00:00:00+00:00",
                      },
                    },
                    "status": "INACTIVE",
                  }
                }
              >
                <div
                  className="emotion-5"
                >
                  High Risk of Violence
                </div>
              </FlagTitle>
              <FlagDescription
                flag={
                  Object {
                    "author": Object {
                      "display": "John Smith",
                      "identifier": undefined,
                      "reference": "https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser",
                      "typeName": "Practitioner",
                    },
                    "category": Object {
                      "coding": Array [
                        Object {
                          "code": "66706",
                          "display": "Violence",
                        },
                      ],
                    },
                    "code": Object {
                      "coding": Array [
                        Object {
                          "code": "56706",
                          "display": "High Risk of Violence",
                        },
                      ],
                    },
                    "id": "580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "end": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                      "start": Object {
                        "kind": "DATE",
                        "value": "2013-01-01T00:00:00+00:00",
                      },
                    },
                    "status": "INACTIVE",
                  }
                }
              >
                <div
                  className="emotion-6"
                >
                  56706 - Violence
                </div>
              </FlagDescription>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-10"
            >
              <PeriodSummary
                period={
                  Object {
                    "end": Object {
                      "kind": "DATE",
                      "value": "2014-02-02T00:00:00+00:00",
                    },
                    "start": Object {
                      "kind": "DATE",
                      "value": "2013-01-01T00:00:00+00:00",
                    },
                  }
                }
              >
                <div
                  className="emotion-2"
                >
                  Jan-01,-2013 to Feb-02,-2014
                </div>
              </PeriodSummary>
              <FlagStatus
                flag={
                  Object {
                    "author": Object {
                      "display": "John Smith",
                      "identifier": undefined,
                      "reference": "https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser",
                      "typeName": "Practitioner",
                    },
                    "category": Object {
                      "coding": Array [
                        Object {
                          "code": "66706",
                          "display": "Violence",
                        },
                      ],
                    },
                    "code": Object {
                      "coding": Array [
                        Object {
                          "code": "56706",
                          "display": "High Risk of Violence",
                        },
                      ],
                    },
                    "id": "580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "end": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                      "start": Object {
                        "kind": "DATE",
                        "value": "2013-01-01T00:00:00+00:00",
                      },
                    },
                    "status": "INACTIVE",
                  }
                }
              >
                <div
                  className="emotion-9"
                >
                  Inactive
                </div>
              </FlagStatus>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </FlagSummary>
  </div>
</Sut>
`;
